### PR TITLE
Add "Remote Persistent Workers"

### PR DESCRIPTION
**Pull request by ulfjack**

### CODE REVIEW

File: designs/2021-03-06-remote-persistent-workers.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+---
+created: 2021-03-06
+last updated: 2021-03-06
+status: To be reviewed
+reviewers:
+  - larsrc-google
```
Comment (lberki) [inline designs/2021-03-06-remote-persistent-workers.md:6]: Mind adding github.com/coeuvre (chiwang@google.com) as a reviewer?

**Review by lberki — APPROVED** [review]

**Review by lberki — CHANGES_REQUESTED** [review]
Mind adding github.com/coeuvre (chiwang@google.com) as a reviewer also?

File: designs/2021-03-06-remote-persistent-workers.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+value to indicate that an input file is a tool input. As of 2021-03-06, our
+prototype only supports non-multiplex, binary protobuf persistent workers, so
+the service assumes that the presence of node properties indicates support for
+this.
+
+In addition, our prototype implementation in Bazel sets `persistentWorkerKey`
+as a platform option, with the value being the computed worker key. This is used
+by the remote execution scheduler to route actions to a matching worker machine.
+
+It would be trivial to extend the prototype to set a `persistentWorkerProtocol`
```
Comment (larsrc-google) [inline designs/2021-03-06-remote-persistent-workers.md:89]: The worker key already contains the protocol format used and whether it's (supposed to be) a multiplex worker. Do you ignore those?

**Comment by ulfjack** [comment]
Added @coeuvre as a reviewer.

**Review by lberki — APPROVED** [review]

**Pull request merged**